src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet45.cs

```diff
@@ -122,35 +123,46 @@ internal IList<Claim> CreateClaims(string issuer)
 
         internal IList<Claim> Claims(string issuer)
         {
-            if (_claims != null)
-                return _claims;
-
-            _claims = new List<Claim>();
-
-            if (!RootElement.HasValues)
-                return _claims;
-
-            // there is some code redundancy here that was not factored as this is a high use method. Each identity received from the host will pass through here.
-            foreach (var entry in RootElement)
+            if (_claims == null)
             {
-                if (entry.Value == null)
-                {
-                    _claims.Add(new Claim(entry.Key, string.Empty, JsonClaimValueTypes.JsonNull, issuer, issuer));
-                    continue;
-                }
-
-                if (entry.Value.Type is JTokenType.String)
+                lock (_claimsLock)
                 {
-                    var claimValue = entry.Value.ToObject<string>();
-                    _claims.Add(new Claim(entry.Key, claimValue, ClaimValueTypes.String, issuer, issuer));
-                    continue;
-                }
-
-                var jtoken = entry.Value;
-                if (jtoken != null)
-                {
-                    AddClaimsFromJToken(_claims, entry.Key, jtoken, issuer);
-                    continue;
+                    if (_claims == null)
```

Comment:
two null checks? Are they needed?

Reply:
yes, this is a standard double checked lock